Client/mods/deathmatch/logic/CClientModelManager.h

```diff
@@ -23,24 +24,23 @@ class CClientModelManager
     friend class CClientModel;
 
 public:
-    CClientModelManager(class CClientManager* pManager);
+    CClientModelManager(class CClientManager* pManager) {}
     ~CClientModelManager(void);
 
     void RemoveAll(void);
 
-    void Add(CClientModel* pModel);
-    bool Remove(CClientModel* pModel);
+    void Add(const std::shared_ptr<CClientModel>& pModel);
+    bool Remove(const std::shared_ptr<CClientModel>& pModel);
 
     int GetFirstFreeModelID(void);
 
-    CClientModel* FindModelByID(int iModelID);
-
-    std::vector<CClientModel*> CClientModelManager::GetModelsByType(const eClientModelType type, const unsigned int minModelID = 0);
+    std::shared_ptr<CClientModel> FindModelByID(int iModelID);
 
+    std::vector<std::shared_ptr<CClientModel>> GetModelsByType(eClientModelType type, const unsigned int minModelID = 0);
 
     void DeallocateModelsAllocatedByResource(CResource* pResource);
 
 private:
-    CClientModel* m_Models[MAX_MODEL_ID];
-    unsigned int m_modelCount = 0;
+    std::shared_ptr<CClientModel> m_Models[MAX_MODEL_ID];
```

Comment:
Perhaps we could use `std::array` instead?
All of it's elements are default initalized, so we woudn't need to null it out in the ctor.

Reply:
An array of shared_ptr is initialized to nullptr by default. The standard guarantees it.

Reply:
Ah, I thought we still have a `memset` for the array. My bad.